src/ibek/entity_factory.py

```diff
@@ -23,18 +26,35 @@
 class EntityFactory:
     def __init__(self) -> None:
         """
-        A class to create `Entity` models from `EntityModel`s.
+        A class to create `Entity` types from `EntityModel` instances.
+
+        To understand this class be aware that these are equivalent:-
+          EntityModel instance == Entity class.
+        i.e. when we instantiate an EntityModel we are creating a new
+        dynamic Entity class.
+
+        EntityModel instantiation happens when:-
+
+        1. Deserializing a support module yaml file
+        2. calling make_entity_model() in ibek_builtin.repeat.py
+
+        Instantiating an Entity itself is the step that happens when
+        deserializing a ioc.yaml. Entities in turn know how to make database
+        and startup script entries for IOC instance.
+
+        I have replaced all references in the code to type[Entity] with
+        EntityModel for consistency.
 
         Created models are stored in `self._entity_models` to lookup when
         resolving nested `SubEntity`s.
         """
-        self._entity_models: dict[str, type[Entity]] = {}
+        self._entity_types: dict[str, EntityModel] = {}
```

Comment:
The type change here makes things clearer, but keep the old name because that already matches.

```suggestion
        self._entity_models: dict[str, EntityModel] = {}
```

Same for similar variable / function names throughout this file.

Reply:
Totally agree. Fixed.